pkg/kubelet/dockershim/helpers_linux.go

```diff
@@ -103,12 +104,17 @@ func (ds *dockerService) updateCreateConfig(
 		// TODO: Can we assume the defaults are sane?
 		rOpts := lc.GetResources()
 		if rOpts != nil {
+			max_pids := config.Linux.MaxPids
+			if max_pids == 0 {
+				max_pids = ds.dockerPIDsLimit
+			}
 			createConfig.HostConfig.Resources = dockercontainer.Resources{
 				Memory:     rOpts.MemoryLimitInBytes,
 				MemorySwap: DefaultMemorySwap(),
 				CPUShares:  rOpts.CpuShares,
 				CPUQuota:   rOpts.CpuQuota,
 				CPUPeriod:  rOpts.CpuPeriod,
+				PidsLimit:  max_pids,
```

Comment:
Should this be a per-container or per-pod limit? If it's per-pod, the Kubelet already does cgroup management of the pod cgroup, so I feel like the Kubelet should just set the limit rather than involving the CRI.

Reply:
interesting, let's wait to hear from others as well. this was super easy to try since `docker stats` gave an easy way for me to try things out ... may need a hand from someone knowledgable.